boxsdk/util/chunked_uploader.py

```diff
@@ -46,17 +47,41 @@ def start(self):
         content_sha1 = self._sha1.digest()
         return self._upload_session.commit(content_sha1=content_sha1, parts=self._part_array)
 
+    def resume(self):
+        """
+        Resumes the process of chunk uploading a file from where upload failed.
+
+        :returns:
+            An uploaded :class:`File`
+        :rtype:
+            :class:`File`
+        """
+        parts = self._upload_session.get_parts()
+        self._part_array = []
+        for part in parts:
+            if self._inflight_part and part['offset'] <= self._inflight_part.offset:
+                self._part_array.append(part)
+            if self._inflight_part and part['offset'] == self._inflight_part.offset:
+                self._inflight_part = None
+            self._part_definitions[part['offset']] = part
+        self._upload()
+        content_sha1 = self._sha1.digest()
+        return self._upload_session.commit(content_sha1=content_sha1, parts=self._part_array)
+
     def _upload(self):
         """
         Utility function for looping through all parts of of the upload session and uploading them.
         """
         while len(self._part_array) < self._upload_session.total_parts:
             next_part = self._inflight_part or self._get_next_part()
             self._inflight_part = next_part
```

Comment:
Now that you have resume functionality, I don't think `_inflight_part` is needed. Instead, you could get the stream position in the constructor, and reset it to the saved position in resume. Then you can just read the chunks as normal.

Reply:
Hey @Jeff-Meadows can you clarify a little more about what you mean by getting the stream position in the constructor and resetting it to the saved position in resume? 

We were operating under the assumption that the stream may not be seakable all the time